awx/main/access.py

```diff
@@ -2092,11 +2092,23 @@ def can_add(self, data):
         if not data:  # So the browseable API will work
             return Organization.accessible_objects(self.user, 'workflow_admin_role').exists()
 
-        return bool(
-            self.check_related('organization', Organization, data, role_field='workflow_admin_role', mandatory=True)
-            and self.check_related('inventory', Inventory, data, role_field='use_role')
-            and self.check_related('execution_environment', ExecutionEnvironment, data, role_field='read_role')
-        )
+        if not self.check_related('organization', Organization, data, role_field='workflow_admin_role', mandatory=True):
+            if data.get('organization', None) is None:
+                self.messages['workflow_job_template'] = [_('An organization is required to create a workflow job template for normal user')]
+                return False
+
+            self.messages['workflow_job_template'] = [_('You do not have workflow_admin_role to the organization')]
```

Comment:
This isn't true any more. `workflow_admin_role` is a role. The core idea of the new system (DAB RBAC) is that it enumerates permissions of a role. So access control only considers permissions, never roles. You get permissions from roles, but policy evaluators should never know what role gave you the permission.

I also don't think that we need this, and I would say it would be more _consistent_ right now to give an unqualified 403 "you do not have permission to perform this action" error message, considering that it was the "add" permission lacked for the "add" action.

Reply:
updated